### ISSUE TITLE
"Compare With" Dropdown chip color issues

### ISSUE THREAD

**Issue by mevionfamos** [visualize-admin/electricity-prices-switzerland#405]
## Description
When selecting multiple operators in the "compare with" dropdown, the text inside the dark chips is unreadable. 
For electricity prices we do not color the chips at all (e.g. https://www.strompreis.elcom.admin.ch/municipality/3661?municipality=3790%2C191%2C243%2C1301%2C1213%2C2553%2C6421) and instead show a small legend for the colors. this might be the easiest solution to avoid this problem. (we already have the legend)  

<img width="851" height="93" alt="Image" src="https://github.com/user-attachments/assets/216a59e4-7ad4-45a2-a63f-30ee5412ae00" />

In addition to this, the colors do not always match the chart below, as discussed [here](https://www.notion.so/interactivethings/251006-Sunshine-Feedback-Barbara-28430da543028017851de8b333c18e97?source=copy_link#28430da543028033a348c0a58666c99b)
Example: https://www.strompreis.abn.elcom.admin.ch/sunshine/operator/335/costs-and-tariffs?tab=netTariffs&compareWith=10839%2C110%2C10773%2C111&viewBy=latest (seems to only affect #412) 

<img width="1448" height="826" alt="Image" src="https://github.com/user-attachments/assets/98ae76db-b191-477e-a32c-c562186669ef" />

## Reproduction Steps
1. Navigate to https://www.strompreis.abn.elcom.admin.ch/sunshine/operator/585/costs-and-tariffs?tab=networkCosts&compareWith=10773%2C11%2C10839%2C107%2C109%2C126%2C11829%2C124%2C11772%2C123%2C11828%2C122%2C127%2C112%2C114%2C137%2C11841%2C129%2C113%2C134&viewBy=latest
3. Observe some of the names are not readable (dark on dark)
4. In addition to not being readable, the colors also do not always match the

## Additional Context
Original feedback from [Notion](https://www.notion.so/interactivethings/Testing-Document-22930da54302804b86b3c7d1dc2ebd04?source=copy_link#23330da54302819cb6f9e4b02eeb1f72):
- Invert the text color where necessary (categorical 05-08 should have a white foreground)
    - Only use `categorical/09 (dark gray)`, and skip `categorical/09 (light gray)`, as it's very similar to 09 and the gridline color
- Make sure that labels wrap too, if necessary (see screenshot)
- URL: https://www.strompreis.ref.elcom.admin.ch/sunshine/operator/600/costs-and-tariffs?tab=networkCosts&compareWith=5%2C6%2C11%2C8&viewBy=latest
- <img width="300"  alt="Image" src="https://github.com/user-attachments/assets/5b51df5e-fd4e-4ccc-ac50-8f8c78db0f70" />
- <img width="300"   alt="Image" src="https://github.com/user-attachments/assets/66d8614f-838c-441c-99ff-abd08b32a030" />
- <img width="300"   alt="Image" src="https://github.com/user-attachments/assets/95b77d72-6930-4942-8bfb-437c9a3c2a35" />

### Test Environment

- [x] [Ref](https://www.strompreis.ref.elcom.admin.ch/)
- [x] [Abn](https://www.strompreis.abn.elcom.admin.ch/)
- [ ] [Prod](https://www.strompreis.elcom.admin.ch/)
- [ ] #456
